src/evmTest/scala/io/iohk/ethereum/vm/utils/EvmTestEnv.scala

```diff
@@ -0,0 +1,142 @@
+package io.iohk.ethereum.vm.utils
+
+import java.io.File
+
+import scala.language.dynamics
+import scala.util.Random
+
+import akka.util.ByteString
+import io.iohk.ethereum.crypto._
+import io.iohk.ethereum.rlp.RLPList
+import io.iohk.ethereum.rlp.RLPImplicits._
+import io.iohk.ethereum.vmrunner.{ABI, MockVmInput}
+import io.iohk.ethereum.vmrunner.Utils
+import io.iohk.ethereum.{rlp, crypto}
+import io.iohk.ethereum.domain.{Account, Address}
+import io.iohk.ethereum.vm._
+
+object EvmTestEnv {
+  val ContractsDir = new File("target/contracts")
+}
+
+// scalastyle:off magic.number
+trait EvmTestEnv {
+
+  import EvmTestEnv._
+
+  private var contractsAddresses: Map[String, Address] = Map.empty
+  private var contractsAbis: Map[String, Seq[ABI]] = Map.empty
+
+  private var internalWorld = MockWorldState()
+
+  val defaultContractCreator = createAccount(balance = 1000)
+  val defaultSender = createAccount(balance = 1000)
+
+  def world: MockWorldState = internalWorld
+
+  def contract(name: String): Contract = {
+    new Contract(name, contractsAddresses(name))
+  }
+
+  def createAccount(balance: BigInt = 0): Address = {
+    val newAddress = Address(Random.nextLong())
+    internalWorld = world.saveAccount(newAddress, Account.Empty.copy(balance = balance))
+    newAddress
+  }
+
+  def deployContract(name: String,
+                     creatorAddress: Address = defaultContractCreator,
+                     value: BigInt = 0,
+                     gasLimit: BigInt = BigInt(2).pow(256) - 1,
+                     gasPrice: BigInt = 1,
+                     constructorArgs: Seq[Any] = Nil): (ProgramResult[MockWorldState, MockStorage], Contract) = {
+    val creator = world.getAccount(creatorAddress).get
+
+    val contractAddress = {
+      val hash = crypto.kec256(rlp.encode(RLPList(creatorAddress.bytes, creator.nonce)))
+      Address(hash.takeRight(Address.Length))
+    }
+
+    val contractInitCode = Utils.loadContractCodeFromFile(new File(s"$ContractsDir/$name.bin"))
+    val contractAbi = Utils.loadContractAbiFromFile(new File(s"$ContractsDir/$name.abi"))
+
+    val payload = constructorArgs.map(parseArg).foldLeft(contractInitCode)(_ ++ _)
+
+    val tx = MockVmInput.transaction(creatorAddress, payload, value, gasLimit, gasPrice)
+    val bh = MockVmInput.blockHeader
+
+    val context = ProgramContext[MockWorldState, MockStorage](tx, bh, world)
+    val result = VM.run(context)
+
+    contractsAbis += (name -> contractAbi.right.get)
```

Comment:
Minor detail: I now that this doesn't fail in our current tests but should we handle the case when this could fail in case `loadContractAbiFromFile` where to return an error?